### ISSUE TITLE
Feature: optionally output two code blocks, one for light and one for dark color scheme

### ISSUE THREAD

**Issue by hirasso** [expressive-code/expressive-code#36]
When having a site that supports both light and dark color themes, it's handy to have rendered code blocks available in both variations, and show the one for the current color scheme based on the user's settings.

The package [markdown-it-shiki](https://github.com/antfu/markdown-it-shiki#dark-mode) is a good example of how this could be solved. 

The counterpart in `expressive-code` could look like this:

```js
/** @type {import('remark-expressive-code').RemarkExpressiveCodeOptions} */
const remarkExpressiveCodeOptions = {
  theme: {
    light: 'min-light',
    dark: 'min-dark'
  }
}
```

**Comment by hippotastic** [comment]
Yes, this is also on my roadmap!

This is also the reason why all CSS styles get prefixed automatically, which makes it easy to output CSS styles for multiple themes without them interfering with each other.

What you can do right now as a workaround (until the multi-theme feature is released) is to use the `styleOverrides` config options in combination with your own site-wide theme-dependent CSS variables. As long as you're not attempting to change code block styles from a very dark to a very light background, this works well. One example is Astro Docs where we use this method.

**Comment by hirasso** [comment]
I would have been surprised if you hadn't though of this already @hippotastic 😄

Thanks for the hint with the style overrides! I think I'll still wait for the actual feature, since our code styles for light vs dark [are very different](https://deploy-preview-119--splendorous-kataifi-9ae281.netlify.app/getting-started/installation/).

**Comment by hippotastic** [comment]
While implementing this, I noticed that there could be a need for more themes than only `light` and `dark` (e.g. a high contrast theme).

Therefore, the `theme` config option now also accepts arrays with any number of themes. The theme's `name` property is used to automatically generate a kebab-cased class name (e.g. `ec-theme-github-dark`) that can be used in your custom site-wide CSS code to only show the code blocks you want.

I just released this feature to NPM in version 0.21.0.
